scripts/get-previous-y-stream.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+this_file="$(basename "${BASH_SOURCE[0]}")"
+
+usage() {
+    >&2 echo "Usage: $this_file <version>
+
+This program prints previous Y-Stream version for a provided <version>.
+
+<version> can be a semantic version, e.g. 3.74.3, and/or the result of make tag, e.g. 3.74.x-nightly-20230224.
+<version> can also include 'v' prefix, e.g. v3.74.3.
+
+Y-Stream is Red Hat term for releases which patch number equals to zero, e.g. 3.73.0, 3.74.0, 4.0.0, 4.1.0.
+This program knows how to subtract one from the minor number of the provided version (e.g. 3.73.0 -> 3.74.0)
+and also knows when major product version was bumped (e.g. 3.74.0 -> 4.0.0)."
+
+    exit 2
+}
+
+main() {
+    if (( $# > 1 )); then
+        >&2 echo "Error: too many command-line arguments provided"
+        exit 1
+    fi
+
+    local version="${1-}"
+
+    if [[ -z "$version" || "$version" == "--help" ]]; then
+        usage
+    fi
+
+    if [[ ! "$version" =~ ^v?([0-9]*)\.([0-9]*)\.(x|[0-9]*)(-.*)?$ ]]; then
```

Comment:
should be
`    if [[ ! "$version" =~ ^v?([0-9]+)\.([0-9]+)\.(x|[0-9]+)(-.+)?$ ]]; then`
`*` also matches 0 repetitions, whereas `+` requires at least 1. A test to differentiate between the two would be to check whether `..-` is a valid version number

Reply:
Good catch. This was a result of me over-editing the regex and not actually reviewing it myself.